CoreFramework/NUnitTestSetup/NUnitTestSetup.cs

```diff
@@ -0,0 +1,64 @@
+﻿using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using OpenQA.Selenium;
+using CoreFramework.DriverCore;
+using AventStack.ExtentReports;
+using CoreFramework.Reporter;
+
+namespace CoreFramework.NUnitTestSetup;
+
+public class NUnitTestSetup
+{
+    // Check why [SetUp] uses InitDriver
+    // Check Add Project Preference 
+    protected IWebDriver? Driver;
+    public WebDriverAction? DriverBaseAction;
+    protected ExtentReports? _extentReport;
+
+
+    [OneTimeSetUp]
+    public void OneTimeSetUp()
+    {
+        HtmlReport.CreateReport();
+
+        // Without this, a suite will not be created
+        // Assign metadata here
```

Comment:
If you want to comment the task which need to be done in the future, then please use the todo comment.
```
// TODO: smt to do
```